x-pack/plugins/security/server/lib/authentication/providers/token.js

```diff
@@ -356,7 +362,30 @@ export class TokenAuthenticationProvider {
       // it's called with this request once again down the line (e.g. in the next authentication provider).
       delete request.headers.authorization;
 
+      // If refresh fails with `400` then refresh token is no longer valid and we should clear session
+      // and redirect user to the login page to re-authenticate.
+      if (getErrorStatusCode(err) === 400 && canRedirectRequest(request)) {
+        this._options.log(
+          ['debug', 'security', 'token'],
+          'Clearing session since both access and refresh tokens are expired.'
```

Comment:
Same comment as above about logging the error message.

Reply:
We log all error messages right at the beginning of the catch :) Or you think just logging `message` isn't enough? Potentially the error can have `body.error.reason` or `body.error.error_description`, but I'm not sure if we want to somehow extract and log these....

Reply:
I think just err.message is fine, I completely missed that we did so above. Please ignore!